src/contracts/examples/intents-example/SwapIntentInvertBidDAppControl.sol

```diff
@@ -0,0 +1,180 @@
+                trustedOpHash: true,
+                invertBidValue: true,
+                exPostBids: false
+            })
+        )
+    { }
+
+    // ---------------------------------------------------- //
+    //                       Custom                         //
+    // ---------------------------------------------------- //
+
+    /*
+    * @notice This is the user operation target function
+    * @dev This function is delegatecalled: msg.sender = Atlas, address(this) = ExecutionEnvironment
+    * @dev It checks that the user has approved Atlas to spend the tokens they are selling and the conditions are met
+    * @param swapIntent The SwapIntent struct
+    */
+    function swap(SwapIntent calldata swapIntent) external payable returns (SwapIntent memory) {
+        require(msg.sender == ATLAS, "SwapIntentDAppControl: InvalidSender");
+        require(_addressPointer() == CONTROL, "SwapIntentDAppControl: InvalidLockState");
+        require(address(this) != CONTROL, "SwapIntentDAppControl: MustBeDelegated");
+
+        address user = _user();
+
+        require(
+            _availableFundsERC20(
+                swapIntent.tokenUserSells, user, swapIntent.maxAmountUserSells, ExecutionPhase.PreSolver
+            ),
+            "SwapIntentDAppControl: SellFundsUnavailable"
+        );
+
+        return SwapIntent({
+            tokenUserBuys: swapIntent.tokenUserBuys,
+            tokenUserSells: swapIntent.tokenUserSells,
+            amountUserBuys: swapIntent.amountUserBuys,
+            maxAmountUserSells: swapIntent.maxAmountUserSells
+        });
+    }
+
+    // ---------------------------------------------------- //
+    //                     Atlas hooks                      //
+    // ---------------------------------------------------- //
+
+    /*
+    * @notice This function is called before a solver operation executes
+    * @dev This function is delegatecalled: msg.sender = Atlas, address(this) = ExecutionEnvironment
+    * @dev It transfers the tokens that the user is selling to the solver
+    * @param solverOp The SolverOperation that is about to execute
+    * @return true if the transfer was successful, false otherwise
+    */
+    function _preSolverCall(SolverOperation calldata solverOp, bytes calldata returnData) internal override {
+        address solverTo = solverOp.solver;
+        if (solverTo == address(this) || solverTo == _control() || solverTo == ATLAS) {
+            revert();
+        }
+
+        SwapIntent memory swapData = abi.decode(returnData, (SwapIntent));
+
+        // The solver must be bidding less than the intent's maxAmountUserSells
+        require(solverOp.bidAmount <= swapData.maxAmountUserSells, "SwapIntentInvertBid: BidTooHigh");
+
+        // Optimistically transfer to the solver contract the amount that the solver is invert bidding
+        _transferUserERC20(swapData.tokenUserSells, solverTo, solverOp.bidAmount);
+
+        return; // success
```

Comment:
We can get rid of these empty returns at the end of pre and postSolver, just to be cleaner